tests/e2e-playwright/tests/conftest.py

```diff
@@ -3,28 +3,192 @@
+        return TypeAdapter(AnyUrl).validate_python(passed_product_url)
+    return TypeAdapter(AnyUrl).validate_python(os.environ["PRODUCT_URL"])
+
+
+@pytest.fixture
+def user_name(request: pytest.FixtureRequest, auto_register: bool, faker: Faker) -> str:
+    if auto_register:
+        faker.seed_instance(random.randint(0, 10000000000))  # noqa: S311
+        return faker.email()
+    if osparc_user_name := request.config.getoption("--user-name"):
+        assert isinstance(osparc_user_name, str)
+        return osparc_user_name
+    return os.environ["USER_NAME"]
+
+
+@pytest.fixture
+def user_password(
+    request: pytest.FixtureRequest, auto_register: bool, faker: Faker
+) -> str:
+    if auto_register:
+        return faker.password(length=12)
+    if osparc_password := request.config.getoption("--password"):
+        assert isinstance(osparc_password, str)
+        return osparc_password
+    return os.environ["USER_PASSWORD"]
+
+
+@pytest.fixture
+def product_billable(request: pytest.FixtureRequest) -> bool:
+    billable = request.config.getoption("--product-billable")
+    return TypeAdapter(bool).validate_python(billable)
+
+
+@pytest.fixture
+def service_test_id(request: pytest.FixtureRequest) -> str:
+    if test_id := request.config.getoption("--service-test-id"):
+        assert isinstance(test_id, str)
+        return test_id
+    return os.environ["SERVICE_TEST_ID"]
+
+
+@pytest.fixture
+def service_key(request: pytest.FixtureRequest) -> str:
+    if key := request.config.getoption("--service-key"):
+        assert isinstance(key, str)
+        return key
+    return os.environ["SERVICE_KEY"]
+
+
+@pytest.fixture
+def auto_register(request: pytest.FixtureRequest) -> bool:
+    return bool(request.config.getoption("--autoregister"))
 
-    yield context.request
+
+@pytest.fixture
+def register(
+    page: Page,
+    product_url: AnyUrl,
+    user_name: str,
+    user_password: str,
+) -> Iterator[Callable[[], AutoRegisteredUser]]:
+    def _do() -> AutoRegisteredUser:
+        print(
+            f"------> Registering in {product_url=} using {user_name=}/{user_password=}"
+        )
+        response = page.goto(f"{product_url}")
+        assert response
+        assert response.ok, response.body()
+        page.get_by_test_id("loginCreateAccountBtn").click()
+        user_email_box = page.get_by_test_id("registrationEmailFld")
+        user_email_box.click()
+        user_email_box.fill(user_name)
+        for pass_id in ["registrationPass1Fld", "registrationPass2Fld"]:
+            user_password_box = page.get_by_test_id(pass_id)
+            user_password_box.click()
+            user_password_box.fill(user_password)
+        with page.expect_response(re.compile(r"/auth/register")) as response_info:
+            page.get_by_test_id("registrationSubmitBtn").click()
+        assert response_info.value.ok, response_info.value.json()
+        return AutoRegisteredUser(user_email=user_name, password=user_password)
```

Comment:
Q: this assumes that the product/deployment allow to register without confirmation email?

Reply:
Yes, this one is mostly useful for local deployment, where you can do this. and also for the next github CI replacement of the e2e tests.